blog/2025-05-28-implementing-unit-tests.md

```diff
@@ -0,0 +1,105 @@
+---
+authors: [Shaheer-Ahmd]
+description: "Implementing unit tests"
+draft: false
+tags: ["eBPF", "ig", "Inspektor Gadget", "Gadget", "inspect", "LFX Mentorship", "unit tests", "testing"]
+title: "Implementing unit tests"
+slug: /2025/05/implementing-unit-tests
+image: /media/2024-05-31-cncf-mentoring-logo.jpg
+---
+
+# Introduction
+
+Hello Inspektors,
+
+I am Shaheer, a CS Major at LUMS, Pakistan, and I have been working under the mentorship of [Burak Ok](https://github.com/burak-ok) and [Qasim Sarfraz](https://github.com/mqasimsarfraz/) for the past 3 months. My project was to implement unit tests for Inspektor Gadget by:  
+
+a) Refactoring the existing code to be more unit-test friendly  
+b) Designing and implementing unit tests from scratch  
+
+For the last part of the mentorship, I also worked on a new gadget, **Snapshot File**.
+
+In this blog, I will be taking you through the challenges and questions that I faced, and how I resolved them.
+
+<!-- truncate -->
+
+## Why are unit tests important
```

Comment:
Also mention that unit tests in general are fast to execute, reducing the development time + faster feedback if something is not working/behaving the right way